### PR TITLE
Remove lazy services

### PR DESCRIPTION
**Pull request by amenophis**
Wait for https://github.com/auth0/auth0-PHP/pull/528 to be merged

### CODE REVIEW

File: tests/Functional/config/default.yml [hunk 1]
```diff
@@ -41,6 +39,6 @@ security:
                 invalidate_session: true
 
 services:
-    app.test.httplug:
-        class: Http\Mock\Client
-        factory: 'Happyr\Auth0Bundle\Tests\Functional\MockedClientFactory::create'
+    http.client:
+        class: Http\Client\HttpClient
+        factory: ['Http\Discovery\HttpClientDiscovery', 'find']
```
Comment (Nyholm) [inline tests/Functional/config/default.yml:44]: Why this change?

File: composer.json [hunk 2]
```diff
@@ -11,7 +11,7 @@
     ],
     "require": {
         "php": ">=7.4",
-        "auth0/auth0-php": "^8.0@beta",
+        "auth0/auth0-php": "dev-8.x-refactor-constructors",
```
Comment (Nyholm) [inline composer.json:14]: We can update the constraint to require beta 2

Comment (amenophis) [reply composer.json:14]: Before the release ?

File: tests/Functional/config/default.yml (hunk 1 above)
Comment (amenophis) [reply tests/Functional/config/default.yml]: Reverted

File: composer.json (hunk 2 above)
Comment (Nyholm) [reply composer.json:14]: Sure.

Comment (Nyholm) [reply composer.json:14]:
```suggestion
        "auth0/auth0-php": "^8.0@beta2",
```

File: composer.json [hunk 3]
```diff
@@ -11,7 +11,7 @@
     ],
     "require": {
         "php": ">=7.4",
-        "auth0/auth0-php": "^8.0@beta",
+        "auth0/auth0-php": "^8.0@beta2",
```
Comment (Nyholm) [inline composer.json:14]:
```suggestion
        "auth0/auth0-php": "^8.0.0-BETA2",
```

**Comment by Nyholm** [comment]
Thank you
